utils/text/operations.cpp

```diff
@@ -46,25 +49,33 @@ text::escape_xml(const std::string& in)
 {
     std::ostringstream quoted;
 
-    const char* delims = "\"&<>'";  // Keep in sync with 'switch' below.
-    std::string::size_type start_pos = 0;
-    std::string::size_type last_pos = in.find_first_of(delims);
-    while (last_pos != std::string::npos) {
-        quoted << in.substr(start_pos, last_pos - start_pos);
-        switch (in[last_pos]) {
-        case '"':  quoted << "&quot;"; break;
-        case '&':  quoted << "&amp;"; break;
-        case '<':  quoted << "&lt;"; break;
-        case '>':  quoted << "&gt;"; break;
-        case '\'': quoted << "&apos;"; break;
-        default:   UNREACHABLE;
+    for (std::string::const_iterator it = in.begin();
+         it != in.end(); ++it) {
+        unsigned char c = (unsigned char)*it;
+        if (c == '"') { 
+            quoted << "&quot;";
+        } else if (c == '&') { 
+            quoted << "&amp;";
+        } else if (c == '<') { 
+            quoted << "&lt;";
+        } else if (c == '>') { 
+            quoted << "&gt;";
+        } else if (c == '\'') { 
+            quoted << "&apos;";
```

Comment:
Why bother respecting all these entities? Why not just use the decimal ASCII value in all cases? The latter would keep the code simpler, and the resulting XML wouldn't depend on the definition of the entities, which are not available in a "plain" XML document.